docs/guides/command_handling.mdx

```diff
@@ -0,0 +1,734 @@
+    // different prefixes for different messages, for example you might want the bot to not require
+    // a prefix when in direct messages. In that case, you might provide a function like this:
+    // ```dart
+    // prefix: (message) {
+    //   if (message.startsWith('!')) {
+    //     return '!';
+    //   } else if (message.guild == null) {
+    //     return '';
+    //   }
+    // }
+    //```
+    //
+    // In this case, we always return `!` meaning that the prefix for all messages will be `!`.
+    prefix: (message) => '!',
+
+    // The `guild` parameter determines what guild slash commands will be registered to by default.
+    //
+    // This is useful for testing since registering slash commands globally can take up to an hour,
+    // whereas registering commands for a single guild is instantaneous.
+    //
+    // If you aren't testing or want your commands to be registered globally, either omit this
+    // parameter or set it to `null`.
+    guild: Snowflake(Platform.environment['GUILD']!),
+
+    // The `options` parameter allows you to specify additional configuration options for the
+    // plugin.
+    //
+    // Generally you can just omit this parameter and use the defaults, but if you want to allow for
+    // a specific behaviour you can include this parameter to change the default settings.
+    //
+    // In this case, we set the option `logErrors` to `true`, meaning errors that occur in commands
+    // will be sent to the logger. If you have also added the `Logging` plugin to your client, these
+    // will then be printed to your console.
+    // `true` is actually the default for this option and was included here for the sake of example.
+    options: CommandsOptions(
+      logErrors: true,
+    ),
+  );
+
+  // Next, we add the commands plugin to our client:
+  client.registerPlugin(commands);
+
+
+  // Finally, we tell the client to connect to Discord:
+  client.connect();
+  //highlight-end
+}
+```
+
+## Registering a command
+
+To register a command, we must first create an instance of the [`ChatCommand`] class.
+This class represents a single command, slash command or not, and once added to the bot will
+automatically be ready to use.
+
+For our first command, let's create the famous `ping` command that'll reply with `pong`.
+
+The first parameter is the command name.
+Command names must obey certain rules, if they don't an error will be thrown.
+Generally, using lower case letters dashes (`-`) instead of spaces or undersoces will avoid any problems.
+
+<br />
+The second parameter is the command description. In traditional command frameworks,
+command descriptions aren't required. However, Discord requires that all slash commands
+have a description, so it is needed in nyxx_commands.
+<br />
+The third parameter is the function that will be executed when the command is ran.
+
+The first parameter to this function must be a [`IContext`].
```

Comment:
`IContext`, while valid, should be replaced with `IChatContext`. Same for the occurrences below.